RI: Follow use case tests

ashiagr opened the pull request:
Parent issue #12821

This PR adds tests for Follow use case.

To test:
Run Test

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ashiagr committed:
Add tests for follow use case

ashiagr committed:
Merge branch 'develop' into issue/follow-usecase-tests

malinajirka commented on WordPress/src/test/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCaseTest.kt:
```diff
@@ -0,0 +1,256 @@
+            whenever(readerPostTableWrapper.isPostFollowed(readerPost)).thenReturn(true)
+            whenever(readerBlogActionsWrapper.followBlogForPost(anyOrNull(), anyBoolean(), anyOrNull())).then {
+                (it.arguments[2] as ActionListener).onActionResult(false)
+                false
```
I believe the code returns true from `followBlogForPost` and invokes `onActionResult(false)` when a request fails. We are returning false even from `followBlogForPost` here - is it intentional?

ashiagr replied:
We do not use the value returned from `followBlogForPost`, instead 
- first emit `PostFollowStatusChanged` for optimistic update [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt#L38)
- then use the return (boolean) value of the [last suspension](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt#L61-L67) point [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/usecases/ReaderSiteFollowUseCase.kt#L45) 

So it seems return value from `followBlogForPost` doesn’t matter in the refactored code. But I see your point and have made a change here: 3912425

malinajirka reviewed (commented):
Thanks @ashiagr! I've left one question but it looks great otherwise. Thanks ;)!

ashiagr committed:
Return true for followBlogForPost

ashiagr committed:
Use un-followed site to test follow site action on request failure

ashiagr committed:
Add missing tests to verify action for followed/un-followed site post

ashiagr commented:
Thanks for the review @malinajirka !

I've addressed your comment. Additionally, added two more commits:
1. 2a5bbed - Use un-followed site to test follow site action on request failure
2. 60c245c - Add few more tests

Ready for another round 🙇‍♀️.

malinajirka reviewed (approved):
Perfect, thanks for all the improvements ;)! LGTM 🚢
